runtime/lua/vim/treesitter/_fold.lua

```diff
@@ -361,4 +361,96 @@ function M.foldexpr(lnum)
   return foldinfos[bufnr].levels[lnum] or '0'
 end
 
+---@package
+---@return { [1]: string, [2]: string[] }[]|string|nil
```

Comment:
Is the `|nil` needed? I don't see any codepath potentially returning `nil`.

Reply:
No, that's a relic from when I thought returning `nil` would be the better failure mode than to fall back to `foldtext`.